cunumeric/array.py

```diff
@@ -2465,6 +2465,65 @@ def diagonal(
                 raise ValueError("Either axis1/axis2 or axes must be supplied")
         return self._diag_helper(offset=offset, axes=axes, extract=extract)
 
+    @add_boilerplate("indices", "values")
+    def put(
+        self, indices: ndarray, values: ndarray, mode: bool = "raise"
+    ) -> None:
+        """
+        Set storage-indexed locations to corresponding values.
+
+        See Also
+        --------
+         numpy.put
+
+        Availability
+        --------
+        Multiple GPUs, Multiple CPUs
+
+        """
+
+        if values.size == 0 or indices.size == 0:
+            return
+
+        if mode not in ("raise", "wrap", "clip"):
+            raise ValueError("clipmode not understood")
+        if mode == "wrap":
+            indices = indices % self.size
+        elif mode == "clip":
+            if np.isscalar(indices):
+                if indices >= self.size:
+                    indices = self.size - 1
+                if indices < 0:
+                    indices = 0
+            else:
+                indices = indices.clip(0, self.size - 1)
+
+        if indices.dtype != np.int64:
+            runtime.warn(
+                "converting index array to int64 type",
+                category=RuntimeWarning,
+            )
+            indices = indices.astype(np.int64)
+
+        if indices.ndim > 1:
+            indices = indices.ravel()
+
+        # in case size of the indices larger than size of the array
+        # some entries in the array will be updated 2 times.
+        # and Legate doesn't guarantee the order in which values
+        # are updated
+        if indices.size > self.size:
+            runtime.warn(
```

Comment:
In general, this warning can be applied to any case when there are duplicated indices. should I add the check for it?
something like:
```
indices_tmp = np.unique (indices)
if indices_tmp.size < indices.size:
    runtime.warn(
```